### PR TITLE
feat: Issue #660 message contract prevention system

### PR DESCRIPTION
**Pull request by ms2sato**
## Summary
Implements comprehensive prevention system for Issue #660 class bugs through semantic contract enforcement between message content (data) and terminal operations (actions).

### Core Components
- **Branded Types**: `MessageContent` / `SubmitKeystroke` for type safety
- **Contract Validation**: `validatePurity()` prevents data/operation mixing  
- **Separation of Concerns**: `MessageContentProcessor` (data) / `PTYOperationExecutor` (operations)
- **Newline Normalization**: Safe `\r\n`/`\r` → `\n` conversion at input stage

### Test Coverage
- **200+ test cases** covering contract violations, integration scenarios, edge cases
- **Issue #660 exact reproduction prevention** verification
- **Comprehensive boundary testing** for all documented scenarios

### Q1: Similar existing mechanism?
**✅ Extension** - Builds upon existing `pty-message-injection-service.ts` ad-hoc fix, elevating to architectural pattern with type safety and contract enforcement. Complementary, not duplicate.

### Q2: Invocation documented in canonical procedure?
**⏳ Not documented yet** - New architectural pattern, first-time introduction. Usage documented in code comments and tests. Procedure documentation appropriate as post-PR follow-up.

### Q4: Full lifecycle documented?  
**✅ Not applicable** - No new file types/directories, existing file modifications only.

## Test Plan
- [x] All existing tests pass
- [x] New prevention system tests: 304 pass (shared) + 1061 pass (server services)
- [x] typecheck: All packages clean
- [x] Integration scenarios: Message splitting prevention verified
- [x] Contract violation detection: Runtime protection confirmed

## Breaking Changes
None. Additive architectural pattern that enhances existing message processing without changing public APIs.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/shared/src/types/message-contracts.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+   */
+  extract: (keystroke: SubmitKeystroke): string => keystroke as string
+}
+
+/**
+ * Type guard to check if a string represents message content.
+ */
+export function isMessageContent(value: unknown): value is MessageContent {
+  return typeof value === 'string'
+}
```
[the remark this answers is not in the document]
Comment (ms2sato) [reply packages/shared/src/types/message-contracts.ts:95]: Resolved in ab00a4f: isMessageContent now checks `typeof value === 'string' && !value.includes('\r')` so isMessageContent('\r') returns false, matching the validatePurity contract.

File: packages/server/src/services/pty-operation-executor.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    // Step 1: Send message content (data)
+    const preservedContent = MessageContentProcessor.extractContent(content)
+    ptyWrite(preservedContent)
+
+    // Step 2: Execute submit operation (action) - SEPARATELY
+    if (shouldSubmit) {
+      const submitKeystroke = PTYOperationExecutor.createSubmitOperation()
+      PTYOperationExecutor.executeSubmit(submitKeystroke, ptyWrite)
+    }
+  }
```
[the remark this answers is not in the document]
Comment (ms2sato) [reply packages/server/src/services/pty-operation-executor.ts:41]: Resolved in ab00a4f: injectMessage now calls PTYOperationExecutor.validatePTYWrite(ptyWrite) after validatePurity. The dedicated helper is no longer bypassed.

File: packages/server/src/services/__tests__/message-contract-integration.test.ts [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+      const content = MessageContentProcessor.process('test')
+
+      expect(() => {
+        PTYOperationExecutor.injectMessage(content, null as any)
+      }).toThrow()
+
+      expect(() => {
+        PTYOperationExecutor.injectMessage(content, 'not a function' as any)
+      }).toThrow()
+    })
```
[the remark this answers is not in the document]
Comment (ms2sato) [reply packages/server/src/services/__tests__/message-contract-integration.test.ts:90]: Resolved in ab00a4f: assertion strengthened to toThrow(/PTY write function is required/) so the test fails loudly if the explicit guard is removed.
